packages/navigator/src/browser/navigator-filter.ts

```diff
@@ -60,14 +60,10 @@ export class FileNavigatorFilter {
         this.emitter.fire(undefined);
     }
 
-    protected onPreferenceChanged(event: PreferenceChangeEvent<FileNavigatorConfiguration>): void {
-        let hasChanged = false;
+    protected onFilesPreferenceChanged(event: PreferenceChangeEvent<FileSystemConfiguration>): void {
         const { preferenceName, newValue } = event;
-        if (preferenceName === 'navigator.exclude') {
+        if (preferenceName === FILES_EXCLUDE_PREFERENCE) {
             this.filterPredicate = this.createFilterPredicate(newValue as FileNavigatorFilter.Exclusions | undefined || {});
-            hasChanged = true;
-        }
-        if (hasChanged) {
```

Comment:
We do not want to refresh the tree on all filesystem preference changes. See here: https://github.com/theia-ide/theia/blob/ba4c62a63bd5a3c3d89e97ca6ee36a2c37da75fd/packages/navigator/src/browser/navigator-tree.ts#L31

In other words, when I change `files.watcherExclude`, I guess we do not need to refresh the navigator. Did you spot a bug?

Reply:
If you change `files.watcherExclude`, it won't do refresh the navigator because in line 65 I check if `files.exclude` preference was  changed.
`if (preferenceName === FILES_EXCLUDE_PREFERENCE) `
@kittaakos maybe I misunderstood you

Reply:
No, I was just blind. Sorry for the noise.